$NetBSD$

Fix ``all'' and ``install'' targets of upstream Makefile:

* fixes to make them work with pkgsrc (s/PREFIX/BINPREFIX, BSD_INSTALL_...)
* disable building and installing of demonstrational programs if not desired
* disable running tests if not desired

--- Makefile.orig	2011-06-21 08:59:00.000000000 +0000
+++ Makefile
@@ -1,29 +1,36 @@
-
 DESTDIR=/usr/local
-PREFIX=polarssl_
+BINPREFIX=polarssl_
 
 .SILENT:
 
 all:
 	cd library  && $(MAKE) all && cd ..
+ifneq ($(ONLY_LIB),yes)
 	cd programs && $(MAKE) all && cd ..
+endif
+ifeq ($(RUN_TESTS),yes)
 	cd tests    && $(MAKE) all && cd ..
+endif
 
 install:
-	mkdir -p $(DESTDIR)/include/polarssl
-	cp -r include/polarssl $(DESTDIR)/include
-	
-	mkdir -p $(DESTDIR)/lib
-	cp library/libpolarssl.* $(DESTDIR)/lib
+	$(BSD_INSTALL_DATA_DIR) $(DESTDIR)$(PREFIX)/include/polarssl
+	$(BSD_INSTALL_DATA) include/polarssl/* $(DESTDIR)$(PREFIX)/include/polarssl
 	
-	mkdir -p $(DESTDIR)/bin
+	$(BSD_INSTALL_LIB_DIR) $(DESTDIR)$(PREFIX)/lib
+	$(BSD_INSTALL_DATA) library/libpolarssl.a $(DESTDIR)$(PREFIX)/lib/libpolarssl.a
+	$(BSD_INSTALL_LIB) library/libpolarssl.so $(DESTDIR)$(PREFIX)/lib/libpolarssl.so.0
+	cd $(DESTDIR)$(PREFIX)/lib; ln -sf libpolarssl.so.0 libpolarssl.so
+
+ifneq ($(ONLY_LIB),yes)
+	$(BSD_INSTALL_PROGRAM_DIR) $(DESTDIR)$(PREFIX)/bin
 	for p in programs/*/* ; do              \
 	    if [ -x $$p ] && [ ! -d $$p ] ;     \
 	    then                                \
-	        f=$(PREFIX)`basename $$p` ;     \
-	        cp $$p $(DESTDIR)/bin/$$f ;     \
+	        f=$(BINPREFIX)`basename $$p` ;  \
+	        $(BSD_INSTALL_PROGRAM) $$p $(DESTDIR)$(PREFIX)/bin/$$f ; \
 	    fi                                  \
 	done
+endif
 
 clean:
 	cd library  && $(MAKE) clean && cd ..
